src/state/slices/targetsSlice.js

```diff
@@ -19,6 +22,12 @@ const targetSlice = createSlice({
         targets: targets.map(({ target }) => target),
       };
     });
+    builder.addMatcher(deleteTargetFulfilled, ({ targets }, action) => {
+      const { originalArgs: targetId = '' } = action.meta?.arg;
+      return {
+        targets: targets.filter(({ id }) => id !== targetId),
```

Comment:
This is ok for this project, but usually after deleting / creating, one would simply fetch again the resources. This is because someone else might have also altered other targets while you were deleting. 
Redux toolkit makes it easier to cache these resources (like targets) with tags. More info:
https://redux-toolkit.js.org/rtk-query/usage/automated-refetching